src/overlays/actors/ovl_En_Wiz_Brock/z_en_wiz_brock.c

```diff
@@ -28,19 +36,109 @@ const ActorInit En_Wiz_Brock_InitVars = {
+        this->despawnCounter++;
+        if ((gGameInfo->data[0x991] + 0x1E) < this->despawnCounter) {
+            Math_ApproachZeroF(&this->dyna.actor.scale.y, (gGameInfo->data[0x992] / 10.0f) + 0.3f,
+                               (gGameInfo->data[0x993] / 10000.0f) + 0.003f);
+            Math_ApproachZeroF(&this->alpha, (gGameInfo->data[0x994] / 10.0f) + 1.0f,
+                               (gGameInfo->data[0x995] / 10.0f) + 35.0f);
+            Math_ApproachF(&this->dyna.actor.scale.x, (gGameInfo->data[0x996] / 100.0f) + 0.02f,
+                           (gGameInfo->data[0x997] / 100.0f) + 0.2f, (gGameInfo->data[0x998] / 1000.0f) + 0.002f);
+            this->dyna.actor.scale.z = this->dyna.actor.scale.x;
+            if (this->dyna.actor.scale.y < 0.001f) {
+                Actor_MarkForDeath(&this->dyna.actor);
+            }
+        }
+    }
+}
+
+void EnWizBrock_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnWizBrock* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void EnWizBrock_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnWizBrock* this = THIS;
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+    func_8012C2DC(globalCtx->state.gfxCtx);
 
-extern UNK_TYPE D_060010E8;
-extern UNK_TYPE D_06001690;
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wiz_Brock/EnWizBrock_Init.s")
+    Matrix_InsertTranslation(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
+                             0);
+    Matrix_Scale(this->dyna.actor.scale.x, this->dyna.actor.scale.y, this->dyna.actor.scale.z, 1);
+    if (this->dyna.actor.colChkInfo.health != 0) {
+        Scene_SetRenderModeXlu(globalCtx, 0, 1);
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetEnvColor(POLY_OPA_DISP++, 255, 255, 255, 255);
+        Gfx_DrawDListOpa(globalCtx, gWizzrobePlatform);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wiz_Brock/EnWizBrock_Destroy.s")
+    } else {
+        Scene_SetRenderModeXlu(globalCtx, 1, 2);
+        gDPPipeSync(POLY_XLU_DISP++);
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, 255, (s8)this->alpha);
+        Gfx_DrawDListXlu(globalCtx, gWizzrobePlatform);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wiz_Brock/func_80A490E4.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wiz_Brock/func_80A490FC.s")
+    if (this->platformType != PLATFORM_TYPE_INACTIVE) {
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        AnimatedMat_Draw(globalCtx, Lib_SegmentedToVirtual(&gWizzrobePlatformTexAnim));
+        gDPPipeSync(POLY_XLU_DISP++);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0x80, 0x80, 255, 255, 255, 255);
+        if (this->platformType == PLATFORM_TYPE_FIRE) {
+            gDPSetEnvColor(POLY_XLU_DISP++, 255, 00, 100, (s8)this->alpha);
+        } else {
+            gDPSetEnvColor(POLY_XLU_DISP++, 50, 00, 255, (s8)this->alpha);
```

Comment:
These s8s should be s16s to avoid signed overflow, which is UB. It does still match, I checked the scratch.